cmd/nitro/nitro.go

```diff
@@ -934,3 +941,80 @@ func testUpdateTxIndex(chainDb ethdb.Database, chainConfig *params.ChainConfig)
 	}
 	log.Info("Tx lookup entries written")
 }
+
+type LiveNodeConfig struct {
+	mutex  sync.RWMutex
+	args   []string
+	config *NodeConfig
+}
+
+func (c *LiveNodeConfig) Get() *NodeConfig {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	return c.config
+}
+
+func (c *LiveNodeConfig) set(config *NodeConfig) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	newConfig := *c.config
+
+	newConfig.Node.Sequencer.MaxBlockSpeed = config.Node.Sequencer.MaxBlockSpeed
+
+	err := initLog(config.LogType, log.Lvl(config.LogLevel))
+	if err != nil {
+		return err
+	}
+	newConfig.LogType = config.LogType
+	newConfig.LogLevel = config.LogLevel
+
+	// TODO
+	// if newConfig.Persistent != config.Persistent {
+	//	log.Info("ignoring persistent config change")
+	//}
+	// if newConfig.Conf != config.Conf || newConfig.Node != config.Node || newConfig.L1 != config.L1 || newConfig.L2 != config.L2 || newConfig.LogLevel != config.LogLevel || newConfig.LogType != config.LogType || newConfig.HTTP != config.HTTP || newConfig.WS != config.WS || newConfig.GraphQL != config.GraphQL || newConfig.Metrics != config.Metrics || newConfig.MetricsServer != config.MetricsServer || newConfig.Init != config.Init {
+	//	return errors.New("some config changes are unsupported")
+	//}
+
+	c.config = &newConfig
+	return nil
+}
+
+func (c *LiveNodeConfig) Start(ctx context.Context) {
+	go func() {
+		for {
+			// TODO
+			// timer := time.NewTimer(c.config.Conf.ReloadInterval)
+			sigusr1 := make(chan os.Signal, 1)
+			signal.Notify(sigusr1, syscall.SIGUSR1)
```

Comment:
subscribing for SIGUSR1 could probably be done once outside the for loop

Reply:
Yeah, I think just moving these 2 lines outside of the loop will work